src/Compiler/Facilities/DiagnosticsLogger.fsi

```diff
@@ -466,15 +464,23 @@ type StackGuard =
 type CompilationGlobalsScope =
     new: diagnosticsLogger: DiagnosticsLogger * buildPhase: BuildPhase -> CompilationGlobalsScope
 
+    /// When dispoed, restores caller's diagnostics logger and build phase.
+    new: unit -> CompilationGlobalsScope
+
     interface IDisposable
 
     member DiagnosticsLogger: DiagnosticsLogger
 
     member BuildPhase: BuildPhase
 
-type CaptureDiagnosticsConcurrently =
-    new: unit -> CaptureDiagnosticsConcurrently
+module MultipleDiagnosticsLoggers =
```

Comment:
```suggestion
module MultipleDiagnosticsLogger =
```
?

Reply:
Yeah, naming is hard. Maybe this could be just `module DiagnosticsLogger`? So we could say `DiagnosticsLogger.Parallel`?
But the name should convey we really do `Async.Parallel` but handling multithreaded diagnostics properly.

Reply:
Yeah why not just `DiagnosticsLogger` actually.

IMO if there are ambitions to do something in parallel (and name is an ambition in that regard) then the handling of multiple threads should be assumed - although this definitely very often mismatches reality :D

Reply:
Now I remembered, there is already:
https://github.com/dotnet/fsharp/blob/0115c93d599a9f9eaec2f974c66c1944231a4913/src/Compiler/Driver/ParseAndCheckInputs.fs#L800

The name `MultipleDiagnosticsLoggers` here indicates the possibility of unifying this code.

Reply:
Oh okay. Maybe then just add a comment about this then.